### DISCUSSION TITLE
Log Level

### DISCUSSION THREAD

**Discussion by ghost** [h3rald/min#80] [Features: On Hold]
Min has some log-levels are: `debug`, `info`, `notice`, `warn`, `error`, `fatal`. And default log-level is `notice`.

However this system, log-leveling system, has been limited by decided loglevels.
Loglevels could not have to be declared as just a keyword that is `warn` example. Loglevels should go on a integer leveling system.
I can say:

`debug` -> `-2`
`info`     -> `-1`
`notice` -> `0`
`warn`   -> `1`
`error`   -> `2`
`fatal`    -> `3`

What is your thinking about that?

**Comment by h3rald** [comment]
Well, internally those levels are implemented like an enum, and they are
actually the standard levels used in Nim as well. I don't remember exactly
what are the values of that enum, only the aliases, and that's why
essentially I decided not to expose integer values, just the labels... are
they 0 to 5? -2 to 3? Just a matter of preference really... they work in
the same way.

On Tue 12 Jan 2021 at 17:45 baykus871 <notifications@github.com> wrote:

> Min has some log-levels are: debug, info, notice, warn, error, fatal. And
> default log-level is notice.
>
> However this system, log-leveling system, has been limited by decided
> loglevels.
> Loglevels could not have to be declared as just a keyword that is warn
> example. Loglevels should go on a integer leveling system.
> I can say:
>
> debug -> -2
> info -> -1
> notice -> 0
> warn -> 1
> error -> 2
> fatal -> 3
>
> What is your thinking about that?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/h3rald/min/discussions/80>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAAJEJI5DHE3UTOZDIYMLDSZR4C3ANCNFSM4V7POWKQ>
> .
>
-- 




Fabio Cevasco
=======================
https://h3rald.com <http://www.h3rald.com>

**Reply by ghost** [reply]
No, i don't think so. I can say a level that has more errority than `fatal` and in this way everybody can say levels that is not existing in Min. So thoughts will bring new levels. I think log level should depend on an integer like level switching.

**Reply by h3rald** [reply]
Well, I'll think about it... I could actually support both as the underlying implementation is integer based anyway!

**Reply by ghost** [reply]
Thanks 👍
